src/router/router.js

```diff
@@ -236,9 +251,14 @@ export const navigate = async function () {
         if (oldView) {
           removeView(previousRoute, oldView, route.transition.out)
         }
+
+        previousRoute = undefined
       }
 
-      state.path = route.path
+      state.rawPath = route.path
+      state.queryParams = route.queryParams
+      state.params = route.params
+      state.path = hash
```

Comment:
something to think about: we could keep `path` and add `hash` (which will contain the populated in route path)